### PR TITLE
[MINOR][PYTHON][DOCS] Update pandas_pyspark.rst

### PR DESCRIPTION
**Pull request by kamelCased**
### What changes were proposed in this pull request?

* 'to'-> 'the'
* 'it requires to create new default index in case' -> 'a new default index is created when'

### Why are the changes needed?

Grammar fix

### Does this PR introduce _any_ user-facing change?

Documentation fix

### How was this patch tested?

No test needed

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Looks fine but mind keeping the PR description template please (https://github.com/apache/spark/blob/master/.github/PULL_REQUEST_TEMPLATE)? Also you might want to enable GitHub Actions in your forked repository to run the build for this PR (see https://github.com/apache/spark/pull/34998/checks?check_run_id=4616648816)

**Comment by kamelCased** [comment]
### What changes were proposed in this pull request?

* 'to'-> 'the'
* 'it requires to create new default index in case' -> 'a new default index is created when'

### Why are the changes needed?

Grammar fix

### Does this PR introduce _any_ user-facing change?

Documentation fix

### How was this patch tested?

No test needed

**Comment by HyukjinKwon** [comment]
ok to test

File: python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst [hunk 1]
```diff
@@ -114,8 +114,8 @@ Spark DataFrame can be a pandas-on-Spark DataFrame easily as below:
    2   8
    3   9
 
-However, note that it requires to create new default index in case pandas-on-Spark DataFrame is created from
-Spark DataFrame. See `Default Index Type <options.rst#default-index-type>`_. In order to avoid this overhead, specify the column
+However, note that a new default index is created when a pandas-on-Spark DataFrame is created from
```
Comment (HyukjinKwon) [inline python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst:117]: I wouldn't try to fix "a pandas-on-Spark DataFrame" and "a Spark DataFrame" here alone .. there are too many instances like this to fix in the whole documentation.

Comment (kamelCased) [reply python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst]: Note and changed. Thanks for your help @HyukjinKwon!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.2.

**Pull request closed without merging**